internal/files/sign.go

```diff
@@ -0,0 +1,110 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package files
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/ProtonMail/gopenpgp/v2/armor"
+	"github.com/ProtonMail/gopenpgp/v2/constants"
+	"github.com/ProtonMail/gopenpgp/v2/crypto"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/logger"
+)
+
+const (
+	signerPrivateKeyfileEnv = "ELASTIC_PACKAGE_SIGNER_PRIVATE_KEYFILE"
+	signerPassphraseEnv     = "ELASTIC_PACKAGE_SIGNER_PASSPHRASE"
+
+	signatureComment = "Signed with elastic-package (using GopenPGP: https://gopenpgp.org)"
+)
+
+type SignOptions struct {
+	PackageName    string
+	PackageVersion string
+}
+
+// VerifySignerConfiguration function verifies if the signer configuration is complete.
+func VerifySignerConfiguration() error {
+	signerPrivateKeyfile := os.Getenv(signerPrivateKeyfileEnv)
+	if signerPrivateKeyfile == "" {
+		return fmt.Errorf("signer private keyfile is required. Please define it with environment variable %s", signerPrivateKeyfileEnv)
+	}
+
+	_, err := os.Stat(signerPrivateKeyfile)
+	if err != nil {
+		return errors.Wrap(err, "can't access the signer private keyfile")
+	}
+
+	signerPassphrase := os.Getenv(signerPassphraseEnv)
```

Comment:
Nit. Not sure if an empty passphrase is a thing, you may want to use `os.LookupEnv` to distinguish unset variable from variable set with an empty string.

Reply:
It's rather a question about the "no password" option. Currently, the implementation doesn't allow for this mode (key with the password required). WDYT?

Reply:
I think is fine, we can revisit the decision later if we find the use case.